lib/elixir/lib/module/types/descr.ex

```diff
@@ -50,6 +51,8 @@ defmodule Module.Types.Descr do
   def reference(), do: %{bitmap: @bit_reference}
   def tuple(), do: %{bitmap: @bit_tuple}
 
+  def boolean(), do: atom([true, false])
```

Comment:
We could reuse your `@boolset` here so all calls to this function could return the same structure instead of generating a new one:
```suggestion
  def boolean(), do: {:union, @boolset}
```

Here is a quick illustration:

```elixir
defmodule Foo do
  @boolset :sets.new(version: 2)
  def bool1, do: {:union, @boolset}
  def bool2, do: {:union, :sets.new(version: 2)}
end

iex> Foo.bool1() === Foo.bool2()
true
iex> :erts_debug.same(Foo.bool1(), Foo.bool1())
true
iex> :erts_debug.same(Foo.bool2(), Foo.bool2())
false
```

Reply:
I see; thanks for the illustration! Pushing that change as well